### PR TITLE
Mes 2906 search results mod1

### PR DESCRIPTION
**Pull request by andrewsetterfield**
## Description

Allow Mod1 tests to be found via the search facility.
https://jira.dvsacloud.uk/browse/MES-2906

User: DESExaminer AM1
current application reference - 20123400011, or complete your own test.
update - new app reference to search with - 20123011011

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)

Screenshot 2020-02-18 at 10 07 50
Screenshot 2020-02-18 at 10 07 58
Screenshot 2020-02-18 at 10 08 06

### CODE REVIEW

File: src/shared/mocks/cat-a-mod1-test-result.mock.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  passCompletion:{
+    passCertificateNumber:'A123456x',
+  },
+  vehicleDetails:{
+    gearboxCategory:'Manual',
+    registrationNumber:'FHKKK',
+  },
+  examinerConducted:10000010,
+  communicationPreferences:{
+    updatedEmail:'Szabi.keresztes@bjss.com',
```
Comment (ammarhaiderbjss) [inline src/shared/mocks/cat-a-mod1-test-result.mock.ts]: :S can we get rid of this please, for @szabikr 's sake!

Comment (andrewsetterfield) [reply src/shared/mocks/cat-a-mod1-test-result.mock.ts]: lol, sure.

File: src/pages/view-test-result/cat-a-mod1/components/debrief-card/debrief-card.ts [hunk 2]
```diff
@@ -22,75 +14,31 @@ import { QuestionResult } from '@dvsa/mes-test-schema/categories/common';
...
-        label: ViewTestResultLabels.control,
-        checked: get(this.data, 'eco.adviceGivenControl', false),
-      },
-      {
-        label: ViewTestResultLabels.planning,
-        checked: get(this.data, 'eco.adviceGivenPlanning', false),
-      },
-    ];
+  ngOnInit() {
+    console.log('this.data', this.data);
```
Comment (ammarhaiderbjss) [inline src/pages/view-test-result/cat-a-mod1/components/debrief-card/debrief-card.ts]: can remove this log now

File: src/pages/view-test-result/cat-a-mod1/components/speed-card/__tests__/speed-card.spec.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        { provide: Config, useFactory: () => ConfigMock.instance() },
+      ],
+    });
+  });
+
+  beforeEach(async(() => {
+  }));
+
+  describe('DOM', () => {
+
```
Comment (ammarhaiderbjss) [inline src/pages/view-test-result/cat-a-mod1/components/speed-card/__tests__/speed-card.spec.ts]: Empty test file

File: src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts [hunk 4]
```diff
@@ -160,10 +158,27 @@ export class ViewTestResultCatAMod1Page extends BasePageComponent implements OnI
       candidateName: getCandidateName(this.testResult.journalData.candidate),
       candidateDriverNumber: this.testResult.journalData.candidate.driverNumber,
       activityCode: this.testResult.activityCode,
-      testOutcome: getTestOutcomeText(this.testResult),
+      testOutcome: this.getTestOutcomeTextAM1(this.testResult),
```
Comment (matthew2564) [inline src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts]: can we use the common getTestOutcomeText here?

File: src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts [hunk 5]
```diff
@@ -78,11 +77,10 @@ export class ViewTestResultCatAMod1Page extends BasePageComponent implements OnI
       .getTestResult(this.applicationReference, this.authenticationProvider.getEmployeeId())
       .pipe(
         map((response: HttpResponse<any>): string => response.body),
-        // todo: PREP-AMOD1 change to CatAMod1UniqueTypes when schema changes are ready
-        map(data => this.testResult = this.compressionProvider.extractTestResult(data) as CatBEUniqueTypes.TestResult),
+        map(data => this.testResult = this.compressionProvider.extractTestResult(data) as TestResultCatAM1Schema),
         tap(() => this.handleLoadingUI(false)),
         catchError((err) => {
-          this.testResult = categoryBETestResultMock;
+          this.testResult = categoryAM1TestResultMock;
```
Comment (matthew2564) [inline src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts]: if an error is thrown and caught at this point, won't it mean that the live application will display this mock data?

Comment (andrewsetterfield) [reply src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts]: was existing code - I'll raise with @ammarhaiderbjssm, but agree. prob shouldn't be there.

File: src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts (hunk 4 above)
Comment (andrewsetterfield) [reply src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts]: removed the local function for this and cast as the common object - seems to work :D

File: src/pages/view-test-result/cat-a-mod1/components/debrief-card/debrief-card.ts (hunk 2 above)
Comment (andrewsetterfield) [reply src/pages/view-test-result/cat-a-mod1/components/debrief-card/debrief-card.ts]: done

File: src/pages/view-test-result/cat-a-mod1/components/speed-card/__tests__/speed-card.spec.ts (hunk 3 above)
Comment (andrewsetterfield) [reply src/pages/view-test-result/cat-a-mod1/components/speed-card/__tests__/speed-card.spec.ts]: writing one now.

File: src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts (hunk 5 above)
Comment (ammarhaiderbjss) [reply src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts]: Can we just get rid of this assignment from B+E and AMod1 and test to see what happens?

File: src/pages/view-test-result/cat-a-mod1/components/speed-card/__tests__/speed-card.spec.ts (hunk 3 above)
Comment (andrewsetterfield) [reply src/pages/view-test-result/cat-a-mod1/components/speed-card/__tests__/speed-card.spec.ts]: removed file and added DOM test into its parent

File: src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts (hunk 5 above)
Comment (andrewsetterfield) [reply src/pages/view-test-result/cat-a-mod1/view-test-result.cat-a-mod1.page.ts]: will look separately from the other comments.

**Review by ammarhaiderbjss — CHANGES_REQUESTED** [review]
Looks awesome, couple of changes -
- Remove fallback to using mock test result
- Hide the 'Vehicle dimensions' row for Mod1

**Review by ammarhaiderbjss — APPROVED** [review]

**Review by matthew2564 — APPROVED** [review]

**Pull request merged**